pymeasure/instruments/attocube/anc300.py

```diff
@@ -0,0 +1,228 @@
+        """ add axis id to a command string at the correct position after the
+        initial command, but before a potential value
+
+        :param command: command string
+        :returns: command string with added axis id
+        """
+        cmdparts = command.split()
+        cmdparts.insert(1, self.axis)
+        return ' '.join(cmdparts)
+
+    def ask(self, command, **kwargs):
+        return self.controller.ask(self._add_axis_id(command), **kwargs)
+
+    def write(self, command, **kwargs):
+        return self.controller.write(self._add_axis_id(command), **kwargs)
+
+    def values(self, command, **kwargs):
+        return self.controller.values(self._add_axis_id(command), **kwargs)
+
+    def stop(self):
+        """ Stop any motion of the axis """
+        self.write('stop')
+
+    def move(self, steps, gnd=True):
+        """ Move 'steps' steps in the direction given by the sign of the
+        argument. This method will change the mode of the axis automatically
+        and ground the axis on the end if 'gnd' is True. The method returns
+        only when the movement is finished.
+
+        :param steps: finite integer value of steps to be performed. A positive
+        sign corresponds to upwards steps, a negative sign to downwards steps.
+        :param gnd: bool, flag to decide if the axis should be grounded after
+        completion of the movement
+        """
+        self.mode = 'stp'
+        # perform the movement
+        if steps > 0:
+            self.stepu = steps
+        elif steps < 0:
+            self.stepd = abs(steps)
+        else:
+            pass  # do not set stepu/d to 0 since it triggers a continous move
+        # wait for the move to finish
+        self.write('stepw')
+        if gnd:
+            self.mode = 'gnd'
+
+    def measure_capacity(self):
+        """ Obtains a new measurement of the capacity. The mode of the axis
+        returns to 'gnd' after the measurement.
+
+        :returns capacity: the freshly measured capacity in nF.
+        """
+        self.mode = 'cap'
+        # wait for the measurement to finish
+        self.ask('capw')
+        return self.capacity
+
+
+class ANC300Controller(Instrument):
+    """ Attocube ANC300 Piezo stage controller with several axes
+
+    :param host: host address of the instrument
+    :param axisnames: a list of axis names which will be used to create
+                      properties with these names
+    :param passwd: password for the attocube standard console
+    :param kwargs: Any valid key-word argument for TelnetAdapter
+    """
+    version = Instrument.measurement(
+           "ver", """ Version number and instrument identification """
+           )
+
+    controllerBoardVersion = Instrument.measurement(
+           "getcser", """ Serial number of the controller board """
+           )
+
+    def __init__(self, host, axisnames, passwd, **kwargs):
+        if 'query_delay' not in kwargs:
+            kwargs['query_delay'] = 0.05
```

Comment:
you could also have that as a kwarg with default argument in __init__ that then gets passed on to AttocubeConsoleAdapter. would you think that clearer (as the default value ends up in the init signature)?

Reply:
I guess your proposed way is slightly more elegant. will change